tf2_geometry_msgs/CMakeLists.txt

```diff
@@ -29,13 +39,18 @@ ament_python_install_package(${PROJECT_NAME}
 add_library(${PROJECT_NAME} INTERFACE)
 target_include_directories(${PROJECT_NAME} INTERFACE
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
-  ${orocos_kdl_INCLUDE_DIRS})
+  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
 target_link_libraries(${PROJECT_NAME} INTERFACE
   ${geometry_msgs_TARGETS}
   orocos-kdl
   tf2::tf2
   tf2_ros::tf2_ros)
+if(TARGET Eigen3::Eigen)
+  # TODO(sloretz) require target to exist when https://github.com/ros2/choco-packages/issues/19 is addressed
```

Comment:
based on this, it seems it won't actually fix the issue on Windows, right?

Reply:
We could also create our own `FindEigen.cmake` module to work around the issue.

Reply:
> based on this, it seems it won't actually fix the issue on Windows, right?

Sorry, I see now that you wrote that in the PR description.
Having a custom find module for eigen is a possible workaround.

Reply:
Yes, and we could possibly even use `eigen_cmake_module` for this.  That said, we know what the issue is (https://github.com/ros2/choco-packages/issues/19), we just need to package a newer version of Eigen for that.  So I'm not sure it is worth it; maybe as part of these fixes, we can spend some time and repackage Eigen.

Reply:
> Yes, and we could possibly even use eigen_cmake_module for this

`eigen_cmake_module` turns Eigen's CMake targets into old-style CMake variables, so it's not much help now that we're using targets ourselves. Once we fix ros2/choco-packages#19 I'd like to remove both this workaround and `eigen_cmake_module` completely.